backend/app/crud_posts.py

```diff
@@ -18,33 +18,43 @@ def get_posts(session: Session, skip: int = 0, limit: int = 10):
     return session.scalars(statement).all()
 
 
-def post_post(session: Session, post: schemas.PostCreate, user_id: int):
-    err = validate_post_fields(post.title, post.caption)
-    err = validate_post_title(session=session, title=post.title)
-
+def post_post(
+    session: Session,
+    # post: schemas.PostCreate,
+    user_id: int,
+    title: str,
+    caption: str,
+    published: bool,
+    img_url: str,
+) -> tuple[models.User, str]:
+    err = validate_post_fields(title, caption)
+    err = validate_post_title(session=session, title=title)
     if err is not None:
         return None, err
 
-    post = models.Post(
-        title=post.title,
-        caption=post.caption,
+    db_post = models.Post(
+        title=title,
+        caption=caption,
         # content=post.content,
+        published=published,
+        img_url=img_url,
+        # img_delete_hash=post.img_delete_hash
```

Comment:
In the response from ImgBB, they supply a `delete_url`. Updating this to reflect that and [this SO link](https://stackoverflow.com/questions/62921251/deleting-images-in-imgbb-via-api) that explains how to go about using that delete url.
```suggestion
        # img_delete_url=post.img_delete_url
```

Reply:
Yeah, that was the idea. Commented because I was focused on making the uploading logic first.